software/deploy/deploy.go

```diff
@@ -409,8 +411,32 @@ func validateIfDagDeployURLCanBeConstructed(deploymentInfo *houston.Deployment)
 }
 
 func getDagDeployURL(deploymentInfo *houston.Deployment) string {
-	c, _ := config.GetCurrentContext()
-	return fmt.Sprintf("https://deployments.%s/%s/dags/upload", c.Domain, deploymentInfo.ReleaseName)
+	// Checks if dagserver URL exists and returns the URL
+	for _, url := range deploymentInfo.Urls {
+		if url.Type == houston.DagServerURLType {
+			logger.Infof("Using dag deploy URL from dagserver: %s", url.URL)
+			return url.URL
+		}
+	}
+
+	// If no dagserver URL is found, we look for airflow URL to detect upload url
+	for _, url := range deploymentInfo.Urls {
+		if url.Type != houston.AirflowURLType {
+			continue
+		}
```

Comment:
initially i added url.Type == houston.AirflowURLType
then start generating the urls 
revive was recommending similar changes what i added currently